src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,140 +1,121 @@
+type Plugin = Plugins[number];
 
-  const handleOnClickOutside = (event: MouseEvent | TouchEvent) => {
-    if (
-      ref.current &&
-      !ref.current.contains(event.target as HTMLElement) &&
-      buttonRef.current &&
-      !buttonRef.current.contains(event.target as HTMLElement)
-    ) {
-      setIsActive(false);
-      setTimeout(() => {
-        if (isActive) {
-          buttonRef.current.focus();
+/**
+ * ```ts
+ * import {Tooltip} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A styled tooltip built on Tippy.js.
+ *
+ * https://atomiks.github.io/tippyjs/
+ * https://github.com/atomiks/tippyjs-react
+ */
+export const Tooltip = ({
+  variant = 'light',
+  align = 'top',
+  className,
+  ...rest
+}: TooltipProps) => {
+  // Hides tooltip when escape key is pressed, following:
+  // https://atomiks.github.io/tippyjs/v6/plugins/#hideonesc
+  const hideOnEsc: Plugin = {
+    name: 'hideOnEsc',
+    defaultValue: true,
+    fn: ({ hide }) => {
+      function onKeyDown(event: KeyboardEvent) {
+        if (event.key === 'Escape') {
+          hide();
         }
-      }, 1);
-    }
+      }
+      return {
+        onShow() {
+          document.addEventListener('keydown', onKeyDown);
+        },
+        onHide() {
+          document.removeEventListener('keydown', onKeyDown);
+        },
+      };
+    },
   };
-
-  /**
-   * Handle Keydown
-   * 1) Remove isActive state and focus on the tooltip button
-   */
-  function onKeyDown(e: any) {
-    if (e.code === ESCAPE_KEYCODE || e.code === TAB_KEYCODE) {
-      setIsActive(false);
-      setTimeout(() => {
-        if (isActive) {
-          buttonRef.current.focus();
-        }
-      }, 1);
-    }
-  }
-  const componentClassName = clsx(styles['tooltip'], className, {
-    [styles['eds-is-active']]: isActive,
-    [styles['tooltip--right']]: align === 'right',
-    [styles['tooltip--below']]: align === 'below',
-    [styles['tooltip--left']]: align === 'left',
-  });
-
   return (
-    <div className={componentClassName} {...other}>
-      <span className={styles['tooltip__trigger']}>
-        <Button
-          variant="bare"
-          hideText
-          iconPosition="after"
-          iconName="question-mark-circle"
-          aria-label={buttonText}
-          text={buttonText}
-          type="button"
-          onClick={onClick}
-          buttonRef={buttonRef}
-        />
-      </span>
-      <span
-        ref={ref}
-        className={styles['tooltip__content']}
-        aria-hidden={!isActive}
-        tabIndex={isActive ? 0 : -1}
-        onKeyDown={onKeyDown}
-      >
-        {children}
-      </span>
-    </div>
+    <Tippy
+      {...rest}
+      className={clsx(
+        styles.tooltip,
+        className,
+        variant === 'light' && styles['tooltip--light'],
+        variant === 'dark' && styles['tooltip--dark'],
+      )}
+      duration={200}
+      placement={align}
+      plugins={[hideOnEsc]}
+    />
   );
 };
+export default Tooltip;
```

Comment:
Noticed the original components were not exporting `default`, is that something we should also do?

Reply:
I don't think it hurts for us to have the extra `Button/index.ts` file, and like the extra flexibility it gives 🙂 if that's not reflected in the plop-templates could you update them as well?

Reply:
@anniehu4 @dierat @jinlee93 This is all good stuff, and happy to adopt whatever ergonomic things like this help the cause. FWIW, I wrote a post that broke down the pros and cons of all these approaches: https://bradfrost.com/blog/post/this-or-that-component-names-index-js-or-component-js/